blog/_posts/2023-01-16-wsl2spackapptainer.md

```diff
@@ -0,0 +1,284 @@
+
+Since there weren't believe to be packages that offered this, we'd need to build stuff from source to get there.
+
+[WSL2](https://learn.microsoft.com/en-us/windows/wsl/install) was used for the Linux environment under Windows, [Spack](https://spack.io/) was used to build software, and [Apptainer](https://apptainer.org/) was used to build a container at the end, that could be taken and used elsewhere, for example on an HPC system.
+
+## Steps
+
+### Setup a WSL2 instance
+
+I'm assuming you have WSL2 enabled on your system, so I'm not covering that step.  It might be as simple as running `wsl --install` in a command prompt, but I promise nothing.
+
+I installed the default [Ubuntu image from the Microsoft Store](https://apps.microsoft.com/store/detail/ubuntu/9PDXGNCFSCZV).
+
+Creating a WSL instance using this is straighforward.  In a command prompt run:
+
+```bash
+wsl --install Ubuntu
+```
+
+Up pops a new terminal, which asks you to set a username and password, and then you're up and running with a working WSL2 instance:
+
+![WSL2 start up](/images/blog/wsl2spackapptainer/wsl2.png)
+
+### Install Spack
+
+Well, before we install Spack, we're going to install some dependencies:
+
+```bash
+sudo apt-get update -y
+sudo apt-get install -y build-essential gfortran
+```
+This is the essential bits that Spack's likely to need - it includes compilers and GNU Make, and I've tacked on a Fortran compiler at the end as I know I'll be needing that later.
+
+Installing Spack itself is just like on any other Linux system:
+
+```bash
+# Get it from github
+git clone https://github.com/spack/spack
+cd spack
+# Switch the the latest release version
+git checkout v0.19.0
+# Source the config file
+. share/spack/setup-env.sh 
+```
+
+That all works cleanly:
+![Install spack](/images/blog/wsl2spackapptainer/installspack.png)
+
+### Check espressomd
+
+We can have a look at the package that exists for espressomd in Spack:
+
+```bash
+$ spack list |grep -i espresso
+py-espresso
+py-espressopp
+quantum-espresso
+```
+
+We have a few options there, and we can look at each one:
+
+```bash
+spack info py-espresso
+```
+
+This shows us it's the one we want, but doesn't mention anything about waLBerla, so it looks like we'll have to sort this bit out ourselves.  You can also look at the full list of dependencies it expects to need using:
+
+```bash
+spack spec py-espresso
+```
+
+It's never a bad thing to do an early check like this anyway, as it triggers Spack to install its necessary bootstrap packages, providing functionality needed by Spack itself.  But sadly, there's no mention of waLBerla.
+
+There's no point tweaking the py-espresso package to use waLBerla yet, given we don't have waLBerla installed.
+
+### Install waLBerla
+
+There isn't a package for waLBerla, but it is a standard CMake package, so should be easy for Spack to handle.
+
+```bash
+spack create https://i10git.cs.fau.de/walberla/walberla/-/archive/v6.1/walberla-v6.1.tar.bz2
```

Comment:
Could we add a comment above briefly saying what `spack create` does?

Reply:
Done.